### ISSUE TITLE
Window title is not updated inside Tmux

### ISSUE THREAD

**Issue by qsmodo** [vifm/vifm#718]
It seems Vifm won't set the window title under Tmux.

1. Open a clean Tmux session with a clean shell (no `.bashrc` file).
    ```
    tmux -f /dev/null
    ```
    Note that `$TERM` is "screen" now.
2. Allow Tmux to set the window title.
    ```
    :set set-titles on
    :set set-titles-string "#{pane_title}"
    ```
    Verify that `printf '\033]0;title\007'` successfully changes the window title.
3. Start a clean Vifm:
    ```
    vifm --no-configs
    ```
4. Enable integration with the terminal multiplexer and allow Vifm to set the window title:
    ```
    :screen
    :set title
    ```
- Expected: The title changes as Vifm changes directories.
- Verified: The title does not change.

Tested under Xterm and St.

**Comment by xaizek** [comment]
Without `set set-titles-string "#{pane_title}"` the title changes (although looks a bit odd).

**Comment by qsmodo** [comment]
Yes, but for me Vifm does not change the title with the directory it's in as in a no-multiplexer session.

**Comment by xaizek** [comment]
Running `:screen` has no effect on the title, only `$TERM` and options make any difference.

Do you see title of a Tmux pane changing?

**Comment by qsmodo** [comment]
No, I don't see the title of the Tmux pane changing in Vifm, although the `printf` commands change the title of the pane.

**Comment by xaizek** [comment]
I'm confused, even with `set-titles-string` things seem to work.

With `TERM=screen` Vifm will probably do (unless terminfo entry defines `tsl` and `fsl` capabilities):
```bash
printf '\033ktitle\033\\'
```
Which should set pane title in both screen and Tmux.

For `TERM=tmux` something like this will be run (because it defines those capabilities):
```bash
printf "%s%s%s" "$(tput tsl)" "my test title" "$(tput fsl)"
```
which sets the title of Xorg window.

Is that how it works for you and what did you expect instead?

**Comment by qsmodo** [comment]
All I see is in the Xorg window title is `0:0:vifm - "hostname"`. The current directory is missing, and changing directories has no effect on the title.

I tested your `printf` commands in a plain Xterm and they have no effects in the title, although it prints to stdout, so at first I'd think it's a terminal thing, but then if I try the `printf` commands in St, they do set the title. 

Still, in both terminals, Vifm in Tmux gives the fixed title `0:0:vifm - "hostname"`, while I expected `/path/ - VIFM`, which works in both terminals outside Tmux. So I guess I should blame Tmux?

**Comment by qsmodo** [comment]
Also, just to recall something I said earlier, `printf '\033]0;title\007'` works in both terminals, in or inside Tmux, but not inside Vifm. In Vifm, according to #587, and I reproduced it, redirecting to the tty works: `:!printf '\033]0;title\007' >/dev/tty`.

**Comment by xaizek** [comment]
> I tested your `printf` commands in a plain Xterm

They aren't meant for Xterm, it uses different ones.

> So I guess I should blame Tmux?

Probably, but also make sure to use correct `$TERM` value. Mine Tmux is 2.1, a different version might operate differently. But generally speaking Vifm communicates with Tmux according to its terminfo entry and has no idea about terminal emulator that you're using. If Tmux doesn't pass escape code to the emulator, Vifm can't do much (except for printing several popular escape codes hoping that some will work and the rest won't affect TUI in any way, which is gambling).

**Comment by qsmodo** [comment]
I see, thank you for the help. Just for the record, my problem is solved with the workaround mentioned in issue 587:

    autocmd DirEnter '*' :!printf '\033]0;'"Vifm - $(dirs)"'\007' > /dev/tty %i

**Comment by xaizek** [comment]
> but not inside Vifm

It works inside Vifm, Vifm just doesn't use this sequence because terminfo specifies another one which is supposed to work.
